### PR TITLE
Omit the base directory from the generated code.

### LINKED ISSUE #462: Extension fields should document which file they came from

### PR DESCRIPTION
**Pull request by swankjesse**
This is dependent on the build directory, which makes it difficult to
do hermetic builds.

Closes #462.

### CODE REVIEW

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 1]
```diff
@@ -422,6 +422,9 @@ public TypeSpec generateMessage(MessageType type) {
       if (!field.documentation().isEmpty()) {
         fieldBuilder.addJavadoc("$L\n", sanitizeJavadoc(field.documentation()));
       }
+      if (field.isExtension()) {
```
Comment (swankjesse) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: FYI @danrice-square

**Comment by JakeWharton** [comment]
Is there any reason to keep the base in `Location` anymore? Seems like we just populate it only to throw it away.

**Comment by JakeWharton** [comment]
LGTM

**Comment by swankjesse** [comment]
The base is still used in error reporting.

**Pull request merged**